plugins/Goals/API.php

```diff
@@ -158,6 +159,7 @@ public function addGoal($idSite, $name, $matchAttribute, $pattern, $patternType,
             'allow_multiple'  => (int)$allowMultipleConversionsPerVisit,
             'revenue'         => $revenue,
             'deleted'         => 0,
+            'event_value_as_revenue' => (int) $useEventValueAsRevenue,
```

Comment:
do we need to add a check that setting can be only true when using a event match attribute?

Reply:
I can add one, but given it shouldn't be used unless it's an event match attribute, I didn't think it would be necessary.

Reply:
it's only to avoid problems if people use the API to create events... they might not know it can only be used for events or so and prevents wrong expectations I guess

Reply:
I see, that makes sense. A clear error message would be useful.